internal/sdkprovider/service/clickhouse/clickhouse_user.go

```diff
@@ -62,17 +78,33 @@ func resourceClickhouseUserCreate(ctx context.Context, d *schema.ResourceData, c
 	serviceName := d.Get("service_name").(string)
 	username := d.Get("username").(string)
 
-	u, err := client.ServiceClickHouseUserCreate(ctx, projectName, serviceName, &clickhouse.ServiceClickHouseUserCreateIn{
-		Name: username,
-	})
+	u, err := client.ServiceClickHouseUserCreate(
+		ctx,
+		projectName,
+		serviceName,
+		&clickhouse.ServiceClickHouseUserCreateIn{
+			Name: username,
+		},
+	)
 	if err != nil {
 		return fmt.Errorf("cannot create ClickHouse user: %w", err)
 	}
 
 	d.SetId(schemautil.BuildResourceID(projectName, serviceName, u.Uuid))
 
-	if err = d.Set("password", u.Password); err != nil {
-		return err
+	passWo := d.GetRawConfig().GetAttr("password_wo")
+	if !passWo.IsNull() {
+		if err = setWriteOnlyPassword(ctx, d, client, projectName, serviceName, u.Uuid); err != nil {
+			return err
+		}
+		return resourceClickhouseUserRead(ctx, d, client)
+	}
+
+	// use generated password
+	if u.Password != nil {
+		if err = d.Set("password", *u.Password); err != nil {
+			return err
+		}
```

Comment:
1. We could rely on `password_wo_version` which if greater than `1` is a solid sine to detect `password_wo`. 
2. I personally would go for `if else` approach, so eventually you have just one `resourceClickhouseUserRead`. Because if we add a new field/logic there are now two branches before "read".